code/datums/martial/plasma_fist.dm

```diff
@@ -46,15 +56,62 @@
 	return
 
 /datum/martial_art/plasma_fist/proc/Plasma(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	var/hasclient = D.client ? TRUE : FALSE
+
 	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
 	playsound(D.loc, 'sound/weapons/punch1.ogg', 50, TRUE, -1)
 	A.say("PLASMA FIST!", forced="plasma fist")
 	D.visible_message("<span class='danger'>[A] hits [D] with THE PLASMA FIST TECHNIQUE!</span>", \
 					"<span class='userdanger'>You're suddenly hit with THE PLASMA FIST TECHNIQUE by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", null, A)
 	to_chat(A, "<span class='danger'>You hit [D] with THE PLASMA FIST TECHNIQUE!</span>")
-	D.gib()
 	log_combat(A, D, "gibbed (Plasma Fist)")
-	return
+	var/turf/Dturf = get_turf(D)
+	D.gib()
+	if(nobomb)
+		return
+	if(!hasclient)
+		to_chat(A, "<span class='warning'>Taking this plasma energy for your </span><span class='notice'>Apotheosis</span><span class='warning'> would bring dishonor to the clan!</span>")
+		new /obj/effect/temp_visual/plasma_soul(Dturf)//doesn't beam to you, so it just hangs around and poofs.
+		return
+	else if(plasma_power >= plasma_cap)
+		to_chat(A, "<span class='warning'>You cannot power up your </span><span class='notice'>Apotheosis</span><span class='warning'> any more!</span>")
+		new /obj/effect/temp_visual/plasma_soul(Dturf)//doesn't beam to you, so it just hangs around and poofs.
+	else
+		plasma_power += plasma_increment
+		to_chat(A, "<span class='nicegreen'>Power increasing! Your </span><span class='notice'>Apotheosis</span><span class='nicegreen'> is now at power level [plasma_power]!</span>")
+		new /obj/effect/temp_visual/plasma_soul(Dturf, A)
+		var/oldcolor = A.color
+		A.color = "#9C00FF"
+		flash_color(A, flash_color = "#9C00FF", flash_time = 3 SECONDS)
+		animate(A, color = oldcolor, time = 3 SECONDS)
+	
+
+/datum/martial_art/plasma_fist/proc/Apotheosis(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	A.say("APOTHEOSIS!!", forced="plasma fist")
+	A.set_species(/datum/species/plasmaman)
+	A.dna.species.species_traits += TRAIT_BOMBIMMUNE
+	A.unequip_everything()
+	A.underwear = "Nude"
+	A.undershirt = "Nude"
+	A.socks = "Nude"
+	A.update_body()
+	var/turf/boomspot = get_turf(A)
+
+	to_chat(A, "<span class='userdanger'>The explosion knocks your soul out of your body!</span>")
+	A.ghostize(FALSE) //prevents... horrible memes just believe me
+
+	A.apply_damage(rand(50,70), BRUTE)
+	log_combat(A, A, "triggered final plasma explosion with size [plasma_power], [plasma_power*2], [plasma_power*4] (Plasma Fist)")
+	message_admins("[key_name_admin(A)] triggered final plasma explosion with size [plasma_power], [plasma_power*2], [plasma_power*4].")
```

Comment:
Move the logging before the ghostize, since it disassociates the key from the mob.